logbook/logbook.go

```diff
@@ -805,6 +805,72 @@ func (book Book) Versions(ctx context.Context, ref dsref.Ref, offset, limit int)
 	return Versions(l, ref, offset, limit), nil
 }
 
+// VersionsWithNotes plays a set of operations for a given log, producing a State struct
+// that describes the current state of a dataset
+func (book Book) VersionsWithNotes(ctx context.Context, ref dsref.Ref, offset, limit int) ([]dsref.VersionInfo, []string, error) {
```

Comment:
I don't like these changes, but didn't manage to think of any other elegant way to handle passing the `op.Note's` up as `DatasetLogItem` doesn't belong this low, yet bubbling up the `op.Log` beats the purpose of having `logbook/logbook.go`.
Open to suggestions on this one.

Reply:
Ah, I see the troublesome situation. `VersionInfo` doesn't have the `commitTitle` and `commitMessage`, and we can't just rely on looking them up from the Dataset store, because for foreign data we won't find them. So we *have* to use the `commitTitle` (op.Note) from the logbook.

I think your first instinct is actually correct. We should move `DatasetLogItem` down to here. Looking at the original refactoring https://github.com/qri-io/qri/commit/4ce3eaa3228aa6f159bb27bbc1dbda7c6ab2c906 which switched `Versions()` over to returning `VersionInfo`: this file used to have a struct called a `DatasetInfo` which have the fields "Ref", "Published", "Timestamp", "CommitTitle", and "Size". That's basically everything stored in the logbook. That means to me we should be doing something similar with having a direct representation at this level. Here are the steps:

```
* move DatasetLogItem from base/ to logbook/
* rename infoFromOp -> itemFromOp
* Versions() returns a []DatasetLogItem
  * commitTitle is assigned, commitMessage will be added later
* rename the accumulator in Versions() to be items instead
* update all callers and tests
```